app/controllers/puppetclasses_controller.rb

```diff
@@ -60,12 +60,13 @@ def parameters
 
   def get_host_or_hostgroup
     # params['host_id'] = 'undefined' if NEW since hosts/form and hostgroups/form has no data-id
-    if params['host_id'] == 'undefined'
+    host_id = params.delete(:host_id)
```

Comment:
Is the `.delete` required and are other parts not relying on host_id?

Reply:
yes, the `host_id` is passed just for the lookup part and isn't permitted by strong params since its a top level param. there is also `params[host][id]` or `params[hostgroup][id]` that should be identical. TBH this code deserves a whole lot of love but that's outside the scope of this pr.